src/tyr/serializers.cc

```diff
@@ -1,6 +1,8 @@
+#include <boost/algorithm/string/replace.hpp>
 #include <boost/property_tree/ptree.hpp>
 #include <cstdint>
 #include <functional>
+#include <rapidjson/allocators.h>
```

Comment:
this is the cause of the preprocessor redefintion. we must only ever include rapidjson stuff via rapidjson_utils.h. if allocators.h isnt in there, add it over there and the preprocessor stuff will go away

Reply:
aaah makes sense!